tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -76,42 +76,51 @@ def _put_license(self):
         assert self.admin.put_license(license).status_code == 200, \
             "License update failed"
 
-        def obtain_license(node):
+        def obtain_configured_license(node):
             lic = self.admin.get_license(node=node)
-            return (lic is not None and lic['loaded'] is True, lic)
+            return (self.admin.is_sample_license(lic), lic)
 
         result = None
         for n in self.redpanda.nodes:
-            resp = wait_until_result(lambda: obtain_license(n),
+            resp = wait_until_result(lambda: obtain_configured_license(n),
                                      timeout_sec=5,
                                      backoff_sec=1)
             assert resp['license'] is not None, "License upload failed!"
             result = resp['license'] if result is None else result
 
         return result
 
+    def _disable_evaluation_period(self):
+        self.redpanda.set_environment(
+            {'__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE': True})
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+
     @skip_fips_mode
     @cluster(num_nodes=3)
-    @matrix(with_license=[
-        True,
-        False,
-    ])
-    def test_get_enterprise(self, with_license):
+    @matrix(with_license=[True, False], with_evaluation_period=[True, False])
+    def test_get_enterprise(self, with_license, with_evaluation_period):
```

Comment:
might consider just marking these ok to fail temporarily and otherwise leaving exactly as is. your changes look fine but it'll be a pain to integrate with https://github.com/redpanda-data/redpanda/pull/23899